transcom/mymove
# INT B-18825 Admin API Approve/Reject/Update requested office user

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-18825)

## Summary

With the incoming missile of office users requesting accounts, we need the functionality to have an endpoint that will allow a MilMove admin to update them, which will be used in the UI for:
- approving the user
- rejecting the user
- updating the user prior to approving/rejecting

This PR does the following:
- adds `PATCH` endpoint for `/requested-office-users/{officeUserId}`
- implements logic that takes in values that can be updated
- I did pull in @cameroncaci's commit containing his implementation of fetching roles in [this PR](https://github.com/transcom/mymove/pull/12199)
- added & updated tests
- updated swagger with relevant data

When calling the API, you can provide data similar to this and it should update. 
```
{
  "firstName": "Daniel",
  "middleInitials": "Bodacious",
  "lastName": "Jordan",
  "telephone": "222-456-7890",
  "roles": [
    {
      "name": "Transportation Invoicing Officer",
      "roleType": "transportation_invoicing_officer"
    },
    {
      "name": "Transportation Ordering Officer",
      "roleType": "transportation_ordering_officer"
    },
    {
      "name": "Contracting Officer",
      "roleType": "contracting_officer"
    },
    {
      "name": "Services Counselor",
      "roleType": "services_counselor"
    },
    {
      "name": "Prime Simulator",
      "roleType": "prime_simulator"
    }
  ],
  "transportationOfficeId": "bb80626f-a953-4832-8c9b-33ca116f9ec7",
  "edipi": "8675309000",
  "otherUniqueId": "thisIsAUniqu3Id",
  "status": "APPROVED"
}
```

### How to test

1. Fire up your `client` & `server` and log in as a MM Admin
2. Using postman or swagger UI
3. Send a `PATCH` request to `/requested-office-users/{officeUserId}`
4. Provide relevant data that can be updated and confirm it updates and persists in the db

## Screenshots
<img width="809" alt="Screenshot 2024-03-13 at 3 02 26 PM" src="https://github.com/transcom/mymove/assets/136510600/08607fc5-7a56-4760-abe5-7855568c723e">
